tests/sdk/evaluation/dp_singleton_query.py

```diff
@@ -30,12 +29,11 @@ def release(self, dataset: object) -> Report:
         """
         private_reader = PrivateReader(dataset[1], dataset[0], self.privacy_params.epsilon)
         query_ast = private_reader.parse_query_string(self.algorithm)
-        srs_orig = private_reader.reader.execute_ast_typed(query_ast)
```

Comment:
Why are internals needed here?

Reply:
I don't expect analysts to ever need to pass in a parsed AST.  Only the evaluator or developers of custom readers will need these.  This simplifies the Reader public interface to have just execute() and execute_df()

Reply:
The internal methods are used on the evaluator for two reasons:  first, to avoid the perf hit of parsing the query and resolving symbols 10,000 times for every run, and second, to allow the cache_exact field to store the exact query results internally (for the accuracy checker).

The first use-case might be relevant to analysts, if there is a scenario that involves calling the same query multiple times per second.

The second use-case is peculiar to the evaluator, and is really not a great design.  We have discussed with James and Michael about switching to a convention where any API that receives infinity for epsilon will, by convention, treat that as a request for a non-private answer.   If we can get a decision on some convention like that, we could eliminate the other use case that requires ASTs.